### PR TITLE
Add support to pass list of functions in postprocess_fn ( Feature discussed in issue #243 )

### PR DESCRIPTION
**Pull request by mainakmanna**
Currently `postprocess_fn` can take only a single function unlike preprocessor. Changes are made to make the behavior similar to the preprocessor - it can take a  list of functions.

### CODE REVIEW

File: t5/data/utils.py [hunk 1]
```diff
@@ -382,9 +382,10 @@ def __init__(self,
         `metric_fn(targets, predictions)` to use during evaluation. By default
         (None), an empty list will be used, resulting in no evaluation on this
         task.
-      postprocess_fn: function, a function that takes in decoded model outputs
-        (strings) and returns a string which is ready for evaluation using the
-        metric functions in `metric_fns`. Can be set to None as a no-op.
+      postprocess_fn: function (or list of functions) that (each) takes in
+        decoded model outputs (strings) and returns a string which is ready
+        for evaluation using the metric functions in `metric_fns`. Can be
+        set to None as a no-op.
```
Comment (adarob) [inline t5/data/utils.py:388]: mention that if multiple functions are provided, they will be applied sequentially

File: t5/data/utils.py [hunk 2]
```diff
@@ -732,6 +731,14 @@ def _feature_config(shape, dtype):
     if self.get_cached_stats(split)["examples"] <= _MAX_EXAMPLES_TO_MEM_CACHE:
       ds = ds.cache()
     return ds
+  def postprocess_fn(self, string, **postprocess_kwargs):
```
Comment (adarob) [inline t5/data/utils.py:734]: add blank line above

**Comment by mainakmanna** [comment]
@googlebot I signed it!

**Comment by mainakmanna** [comment]
> Thank!



> Thank!

Thanks for the feedback comments.

**Comment by adarob** [comment]
Looks like Travis is broken. Please fix.

File: t5/data/utils.py [hunk 3]
```diff
@@ -733,6 +733,15 @@ def _feature_config(shape, dtype):
       ds = ds.cache()
     return ds
 
+  def postprocess_fn(self, string, **postprocess_kwargs):
+    """Returns the processed string after applying postprocess function(s)"""
+    if not hasattr(self._postprocess_fn, "__iter__"):
+      postprocessors = [self._postprocess_fn]
+    for post_fn in postprocessors:
```
Comment (adarob) [inline t5/data/utils.py:740]: postprocessors isn't defined when a list is given causing test failure

**Pull request merged**